agent/api/task.go

```diff
@@ -409,12 +417,30 @@ func (task *Task) dockerHostConfig(container *Container, dockerContainerMap map[
 		return nil, &HostConfigError{err.Error()}
 	}
 
+	// Populate hostConfig
 	hostConfig := &docker.HostConfig{
 		Links:        dockerLinkArr,
 		Binds:        binds,
 		PortBindings: dockerPortMap,
 		VolumesFrom:  volumesFrom,
 	}
+	// Update cgroup parent
+	// TODO: Feature gating
+	if task.CgroupEnabled() {
```

Comment:
This is another good place for platform-specific wrappers instead of a plain feature-gate.